cli/man_main.c

```diff
@@ -0,0 +1,195 @@
+				continue;
+			char *desc = ec_node_desc(str_node);
+			if (desc == NULL)
+				continue;
+
+			if (mode == SYNOPSIS_MODE && !first) {
+				free(desc);
+				continue;
+			}
+
+			printf("%s**%s**", mode == SYNOPSIS_MODE || first ? "" : ", ", desc);
+			first = false;
+			free(desc);
+		}
+	} else if (strcmp(child_type, "seq") == 0) {
+		if (ec_node_get_children_count(child) < 2)
+			return;
+
+		struct ec_node *or_node, *arg_node;
+		unsigned or_refs, arg_refs;
+
+		if (ec_node_get_child(child, 0, &or_node, &or_refs) >= 0) {
+			const char *or_type = ec_node_type(or_node)->name;
+			if (strcmp(or_type, "or") == 0) {
+				bool first = true;
+				for (unsigned i = 0; i < ec_node_get_children_count(or_node); i++) {
+					struct ec_node *str_node;
+					if (ec_node_get_child(or_node, i, &str_node, &refs) < 0)
+						continue;
+					char *desc = ec_node_desc(str_node);
+					if (desc == NULL)
+						continue;
+
+					if (mode == SYNOPSIS_MODE && !first) {
+						free(desc);
+						continue;
+					}
+
+					printf("%s**%s**",
+					       mode == SYNOPSIS_MODE || first ? "" : ", ",
+					       desc);
+					first = false;
+					free(desc);
+				}
+			}
+		}
+
+		if (ec_node_get_child(child, 1, &arg_node, &arg_refs) >= 0) {
+			const char *arg_id = ec_node_id(arg_node);
+			if (arg_id != NULL && strcmp(arg_id, EC_NO_ID) != 0)
+				arg_name = arg_id;
+		}
+	}
+
+	if (arg_name != NULL) {
+		printf(" _");
+		for (const char *p = arg_name; *p; p++)
+			putchar(toupper((unsigned char)*p));
+		printf("_");
+	}
+
+	if (mode == SYNOPSIS_MODE)
+		printf("]\n");
+	else {
+		printf("\n\n");
+		const char *help = ec_dict_get(ec_node_attrs(cmd_node), "help");
+		if (help != NULL)
+			printf("%s\n\n", help);
+	}
+}
+
+static int print_cli_options_from_tree(void) {
+	struct ec_node *options_tree = grcli_options_node();
+	if (options_tree == NULL) {
+		fprintf(stderr, "Error: failed to create options tree\n");
+		return EXIT_FAILURE;
+	}
+
+	printf("# SYNOPSIS\n\n");
+	printf("**grcli**\n");
+
+	for (unsigned i = 0; i < ec_node_get_children_count(options_tree); i++) {
+		struct ec_node *opt_node;
+		unsigned refs;
+		if (ec_node_get_child(options_tree, i, &opt_node, &refs) < 0)
+			continue;
+		print_option_syntax(opt_node, SYNOPSIS_MODE);
+	}
+	printf("...\n\n");
+
+	printf("# OPTIONS\n\n");
+	for (unsigned i = 0; i < ec_node_get_children_count(options_tree); i++) {
+		struct ec_node *opt_node;
+		unsigned refs;
+		if (ec_node_get_child(options_tree, i, &opt_node, &refs) < 0)
+			continue;
+		print_option_syntax(opt_node, OPTION_MODE);
+	}
+
+	ec_node_free(options_tree);
+	return EXIT_SUCCESS;
+}
+
+int print_main_man_page(void) {
+	char title[128];
+	snprintf(title, sizeof(title), "GRCLI 1 \"grout %s\"", GROUT_VERSION);
+	printf("%s\n", title);
+	man_print_title_underline(title);
+	printf("# NAME\n\n");
+	printf("**grcli** -- grout command line interface\n\n");
+
+	if (print_cli_options_from_tree() != EXIT_SUCCESS)
+		return EXIT_FAILURE;
```

Comment:
That is a weird return value for a function. Usually, an error is a negative number and 0 on success. The EXIT_SUCCESS and EXIT_FAILURE constants should only be used in main().